plugins/com.google.cloud.tools.eclipse.util/templates/appengine/web.xml.ftl

```diff
@@ -3,26 +3,6 @@
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="${namespace} ${schemaUrl}"
          version="${servletVersion}">
-<#if servletVersion == "2.5">
```

Comment:
Java 8 runtime still supports servlet 2.5 as well as 3.1. I wonder if we need a separate choice for that?

Reply:
I get that, but there is Servlet 4.0 (2017), and 2.5 is really ancient (2005). As a project creation wizard, I think we should discourage 2.5.

Reply:
OK, we can take it out of the UI for now and put it back if there's demand, but let's leave in options to support it like this file.

Reply:
I restored `compilerVersion` (currently 1.8) and `servletVersion` (currently 3.1) in `pom.xml`.

For `web.xml`, we can git rid of this, because with >=3.0, we won't be setting `<servletMapping>` anymore.

Note that in my original PR, I did try to leave branching possibilities in other places for future changes (like Java 11), as we discussed. `pom.xml` was an exception because I just took `pom.flex.ftl`, which did not have template substitution at all.

Reply:
I'm not sure we can abandon servlet 2.5 so readily. There is a lot of legacy code using this, and a lot of tutorials and books and articles based on this.

Reply:
Alright, we can leave the code to generate Servlet 2.5 stuff.